### PR TITLE
translation: Importing And Exporting Components

### PR DESCRIPTION
**Pull request by JamieLoLo**

### CODE REVIEW

File: src/content/learn/importing-and-exporting-components.md [hunk 1]
```diff
@@ -1,26 +1,26 @@
...
+* 什麼是 root component 檔案
+* 如何 import 和 export 一個 component
+* 何時使用 default 和 named import 和 export
+* 如何從一個檔案 import 和 export 多個 component
+* 如何將 component 拆分為多個檔案
 
 </YouWillLearn>
 
-## The root component file {/*the-root-component-file*/}
+## root component 檔案 {/*the-root-component-file*/}
```
Comment (neighborhood999) [inline src/content/learn/importing-and-exporting-components.md:21]:
```suggestion
## Root component 檔案 {/*the-root-component-file*/}
```

File: src/content/learn/importing-and-exporting-components.md [hunk 2]
```diff
@@ -52,17 +52,17 @@ img { margin: 0 10px 10px 0; height: 90px; }
...
+如果將來你想要改變首頁，在此頁面放入科學書籍列表，或者需要將所有的資料移至其他檔案中。將 `Gallery` 以及 `Profile` 移出 root component 檔案會更加合理。這將會使它們更加模組化，並且可以在其他檔案中複用。你可以透過以下三個步驟拆分 component：
 
-1. **Make** a new JS file to put the components in.
-2. **Export** your function component from that file (using either [default](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Statements/export#using_the_default_export) or [named](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Statements/export#using_named_exports) exports).
-3. **Import** it in the file where you’ll use the component (using the corresponding technique for importing [default](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Statements/import#importing_defaults) or [named](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Statements/import#import_a_single_export_from_a_module) exports).
+1. **創建** 一個新的 JS 檔案來存放該 component。
```
Comment (neighborhood999) [inline src/content/learn/importing-and-exporting-components.md:61]:
```suggestion
1. **建立**一個新的 JS 檔案來存放該 component。
```

File: src/content/learn/importing-and-exporting-components.md [hunk 3]
```diff
@@ -104,82 +104,82 @@ img { margin: 0 10px 10px 0; height: 90px; }
...
 </Sandpack>
 
-Notice how this example is broken down into two component files now:
+請注意此範例是如何將 component 拆分為兩個檔案：
 
 1. `Gallery.js`:
-     - Defines the `Profile` component which is only used within the same file and is not exported.
-     - Exports the `Gallery` component as a **default export.**
+     - 定義了 `Profile` component，該 component 僅在同個檔案中使用，並沒有被 export。
+     - 使用 **default export** 的方式， export `Gallery` component。
```
Comment (neighborhood999) [inline src/content/learn/importing-and-exporting-components.md:111]:
```suggestion
     - 使用 **default export** 的方式，export `Gallery` component。
```

File: src/content/learn/importing-and-exporting-components.md [hunk 4]
```diff
@@ -104,82 +104,82 @@ img { margin: 0 10px 10px 0; height: 90px; }
...
+**當檔案中只需要 export 一個 component 時，人們通常會使用 default export，當檔案包含多個 component 或值需要 export 時，則會使用 named export**。無論你偏好哪種方式，請記得給予 component 以及對應檔案一個有意義的命名。不建議使用未命名的 component，像是 `export default () => {}`，這將導致除錯變得困難。
 
 </DeepDive>
 
-## Exporting and importing multiple components from the same file {/*exporting-and-importing-multiple-components-from-the-same-file*/}
+## 從同一檔案 export 及 import 多個 component {/*exporting-and-importing-multiple-components-from-the-same-file*/}
 
-What if you want to show just one `Profile` instead of a gallery? You can export the `Profile` component, too. But `Gallery.js` already has a *default* export, and you can't have _two_ default exports. You could create a new file with a default export, or you could add a *named* export for `Profile`. **A file can only have one default export, but it can have numerous named exports!**
+如果你只想要展示一個 `Profile`，而非整個圖集。你也可以 export `Profile` component。但是 `Gallery.js` 已經包含 _default_ export，你不能有*兩個* default export。你可以建立一個新檔案以進行 default export，或是你可以將 `Profile` 進行 _named_ export。**同一檔案只能有一個 default export，但可以有多個 named export！**
```
Comment (neighborhood999) [inline src/content/learn/importing-and-exporting-components.md:152]:
```suggestion
如果你只想要顯示一個 `Profile`，而非整個圖集。你也可以 export `Profile` component。但是 `Gallery.js` 已經包含 _default_ export，你不能有*兩個* default export。你可以建立一個新檔案以進行 default export，或是你可以將 `Profile` 進行 _named_ export。**同一檔案只能有一個 default export，但可以有多個 named export！**
```

**Review by neighborhood999 — COMMENTED** [review]
Thanks for your contributions!

**Comment by JamieLoLo** [comment]
Hi, @neighborhood999 

The files have been modified according to the review suggestions. Thank you very much for your feedback!

**Pull request merged**